File: src/main/webapp/js/app/master/receiver.service.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

(function() {

	var services = angular.module('je-master-receiver-service', [ 'ngResource' ]);

	
	services.factory('ReceiverListFactory', [ '$resource', function($resource) {

		return $resource('service/master/receiver', {}, {
			query : {
				method : 'GET',
				isArray : false
			},
			create : {
				method : 'POST'
			}
		});

	} ]);
	
	services.factory('ReceiverEditFactory', [ '$resource', function($resource) {

		return $resource('service/master/receiver/:id', {}, {
			show : {
				method : 'GET',
				params: {id: '@id'}
			},
			update : {
				method : 'PUT',
				params :{id: '@id'}
			},
			remove : {
				method : 'DELETE',
				params :{id: '@id'}
			}
		});

	} ]);
	
})();

File: src/main/java/com/je/spring/rest/master/service/ReceiverService.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.je.spring.rest.master.service;

import com.je.spring.rest.master.model.Receiver;
import java.util.List;
/**
 *
 * @author Prasetyo
 */
public interface ReceiverService {
    public List<Receiver> getAll();
    public Receiver getById(int id);
    public long count();
    public void insert(Receiver receiver);
    public void update(Receiver receiver);
    public void delete(Receiver receiver);
}
